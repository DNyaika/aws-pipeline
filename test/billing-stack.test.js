"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const aws_cdk_lib_1 = require("aws-cdk-lib");
const billing_stack_1 = require("../lib/billing-stack");
const assertions_1 = require("aws-cdk-lib/assertions");
test("Billing Stack", () => {
    const app = new aws_cdk_lib_1.App();
    const stack = new billing_stack_1.BillingStack(app, 'TestStack', {
        budgetAmount: 1,
        emailAddress: "test@example.com",
    });
    const template = assertions_1.Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
});
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiYmlsbGluZy1zdGFjay50ZXN0LmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiYmlsbGluZy1zdGFjay50ZXN0LnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7O0FBQUEsNkNBQWtDO0FBQ2xDLHdEQUFvRDtBQUNwRCx1REFBa0Q7QUFFbEQsSUFBSSxDQUFDLGVBQWUsRUFBRSxHQUFHLEVBQUU7SUFDdkIsTUFBTSxHQUFHLEdBQUcsSUFBSSxpQkFBRyxFQUFFLENBQUM7SUFDdEIsTUFBTSxLQUFLLEdBQUcsSUFBSSw0QkFBWSxDQUFDLEdBQUcsRUFBRSxXQUFXLEVBQUU7UUFDN0MsWUFBWSxFQUFFLENBQUM7UUFDZixZQUFZLEVBQUUsa0JBQWtCO0tBQ25DLENBQUMsQ0FBQztJQUVILE1BQU0sUUFBUSxHQUFHLHFCQUFRLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxDQUFDO0lBQzNDLE1BQU0sQ0FBQyxRQUFRLENBQUMsTUFBTSxFQUFFLENBQUMsQ0FBQyxlQUFlLEVBQUUsQ0FBQztBQUNoRCxDQUFDLENBQUMsQ0FBQyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IEFwcCB9IGZyb20gJ2F3cy1jZGstbGliJztcbmltcG9ydCB7IEJpbGxpbmdTdGFjayB9IGZyb20gJy4uL2xpYi9iaWxsaW5nLXN0YWNrJztcbmltcG9ydCB7IFRlbXBsYXRlIH0gZnJvbSAnYXdzLWNkay1saWIvYXNzZXJ0aW9ucyc7XG5cbnRlc3QoXCJCaWxsaW5nIFN0YWNrXCIsICgpID0+IHtcbiAgICBjb25zdCBhcHAgPSBuZXcgQXBwKCk7XG4gICAgY29uc3Qgc3RhY2sgPSBuZXcgQmlsbGluZ1N0YWNrKGFwcCwgJ1Rlc3RTdGFjaycsIHtcbiAgICAgICAgYnVkZ2V0QW1vdW50OiAxLFxuICAgICAgICBlbWFpbEFkZHJlc3M6IFwidGVzdEBleGFtcGxlLmNvbVwiLFxuICAgIH0pO1xuICAgIFxuICAgIGNvbnN0IHRlbXBsYXRlID0gVGVtcGxhdGUuZnJvbVN0YWNrKHN0YWNrKTtcbiAgICBleHBlY3QodGVtcGxhdGUudG9KU09OKCkpLnRvTWF0Y2hTbmFwc2hvdCgpO1xufSk7Il19