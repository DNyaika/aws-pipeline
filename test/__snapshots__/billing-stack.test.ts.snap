// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Billing Stack 1`] = `
{
  "Resources": {
    "BudgetED13AE41": {
      "Properties": {
        "Budget": {
          "BudgetLimit": {
            "Amount": 1,
            "Unit": "USD",
          },
          "BudgetName": "Monthly Budget",
          "BudgetType": "COST",
          "TimeUnit": "MONTHLY",
        },
        "NotificationsWithSubscribers": [
          {
            "Notification": {
              "ComparisonOperator": "GREATER_THAN",
              "NotificationType": "ACTUAL",
              "Threshold": 100,
            },
            "Subscribers": [
              {
                "Address": "test@example.com",
                "SubscriptionType": "EMAIL",
              },
            ],
          },
        ],
      },
      "Type": "AWS::Budgets::Budget",
    },
  },
}
`;
